webapp/channels/src/components/emoji/emoji_page.tsx

```diff
@@ -25,62 +25,52 @@ type Props = {
     };
 }
 
-export default class EmojiPage extends React.PureComponent<Props> {
-    static defaultProps = {
-        teamName: '',
-        teamDisplayName: '',
-        siteName: '',
-    };
-
-    componentDidMount() {
-        this.updateTitle();
-        this.props.actions.loadRolesIfNeeded(['system_admin', 'team_admin', 'system_user', 'team_user']);
+export default function EmojiPage({teamDisplayName = '', teamName = '', siteName = '', scrollToTop, currentTheme, actions}: Props) {
+    useEffect(() => {
+        updateTitle();
+        actions.loadRolesIfNeeded(['system_admin', 'team_admin', 'system_user', 'team_user']);
         Utils.resetTheme();
-    }
 
-    componentWillUnmount() {
-        Utils.applyTheme(this.props.currentTheme);
-    }
+        return () => {
+            Utils.applyTheme(currentTheme);
+        };
+    }, [actions, currentTheme]);
```

Comment:
This is supposed to run only on component did mount, so let's remove all the dependencies.
```suggestion
    }, []);
```
This may create some lint warnings, but that is ok.